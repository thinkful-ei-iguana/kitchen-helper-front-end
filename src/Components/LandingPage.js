import React from "react";
import { Link } from "react-router-dom";
import homePage from "../../src/Assets/Loggedinmainpage.png";
import LoggingIn from '../../src/Assets/loggingin.png';
import recipeDetail from "../../src/Assets/detailedviewofarecipe.png"


export default class LandingPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = { isLoggedIn: false };
  }

  render() {
    return (
      <div className="LandingPage">
        <header id="LandingPage-Header" className="flex-container">
          <div className="flex-item branding">
          </div>
          <div className="flex-item"></div>
        </header>
        <section id="landingPageDocumentation">
          <div className="site-description">
            <p className="p1">
              Kitchen Helper is a best friend in the kitchen. You have the ability to add
              your own recipes and update them as you need.
            </p>
            <p className="p2">
              Kitchen Helper will be a time and money saver, helping you create
              amazing meals at home while staying organized and never running
              out of that special ingredient you need for your amazing lasagna
              recipe.
            </p>

            <div className="login-signup">
              <div className="LoginBtn">
                <label className="login-label">Login: </label>
                <Link className="login-btn" to="/login">
                  Let's Get Cooking!
                </Link>
              </div>
              <div className="RegistrationBtn">
                <label className="registration-label">Sign Up:</label>
                <Link className="registration-btn" to="/create-account">
                  Come Join the Fun!
                </Link>
              </div>
            </div>
            <h4>Please test out our site using the username of 'guestuser' and the password of Password!2</h4>
            <div className="sampleImages">
              <img src={LoggingIn} className="landingSamples" alt="Logging Into Main Page" id="logging-in" />
              <img src={homePage} className="landingSamples" alt="Home Page" id="logged-in-main-page" />
              <img src={recipeDetail} className="landingSamples" alt="Recipe Detail" id="recipe-details" />
            </div>
          </div>
        </section>
        <section id="Landing-Buttons">
          <hr id="Connect-Divider" />
          <div id="buttons"></div>
        </section>
      </div>
    );
  }
}
