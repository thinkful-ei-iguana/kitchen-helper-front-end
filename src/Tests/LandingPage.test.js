import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import LandingPage from '../Components/LandingPage';


it('should be able to run tests', () => {
  expect(1 + 2).toEqual(3);
})

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter>
    <LandingPage />
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<BrowserRouter>
    <LandingPage />
  </BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
