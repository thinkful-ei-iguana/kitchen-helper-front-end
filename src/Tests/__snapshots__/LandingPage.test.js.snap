// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="LandingPage"
>
  <header
    className="flex-container"
    id="LandingPage-Header"
  >
    <div
      className="flex-item branding"
    />
    <div
      className="flex-item"
    />
  </header>
  <section
    id="landingPageDocumentation"
  >
    <div
      className="site-description"
    >
      <p
        className="p1"
      >
        Kitchen Helper is a best friend in the kitchen. You have the ability to add your own recipes and update them as you need.
      </p>
      <p
        className="p2"
      >
        Kitchen Helper will be a time and money saver, helping you create amazing meals at home while staying organized and never running out of that special ingredient you need for your amazing lasagna recipe.
      </p>
      <div
        className="login-signup"
      >
        <div
          className="LoginBtn"
        >
          <label
            className="login-label"
          >
            Login: 
          </label>
          <a
            className="login-btn"
            href="/login"
            onClick={[Function]}
          >
            Let's Get Cooking!
          </a>
        </div>
        <div
          className="RegistrationBtn"
        >
          <label
            className="registration-label"
          >
            Sign Up:
          </label>
          <a
            className="registration-btn"
            href="/create-account"
            onClick={[Function]}
          >
            Come Join the Fun!
          </a>
        </div>
      </div>
      <h4>
        Please test out our site using the username of 'guestuser' and the password of Password!2
      </h4>
      <div
        className="sampleImages"
      >
        <img
          alt="Logging Into Main Page"
          className="landingSamples"
          id="logging-in"
          src="loggingin.png"
        />
        <img
          alt="Home Page"
          className="landingSamples"
          id="logged-in-main-page"
          src="Loggedinmainpage.png"
        />
        <img
          alt="Recipe Detail"
          className="landingSamples"
          id="recipe-details"
          src="detailedviewofarecipe.png"
        />
      </div>
    </div>
  </section>
  <section
    id="Landing-Buttons"
  >
    <hr
      id="Connect-Divider"
    />
    <div
      id="buttons"
    />
  </section>
</div>
`;
